lang/test/org/partiql/lang/eval/time/TimeTest.kt

```diff
@@ -0,0 +1,177 @@
+        TimeTestCase(hour, min, sec, nano, precision, tz_min, expectedLocalTime, expectedZoneOffset)
+
+    fun case(hour: Int, min: Int, sec: Int, nano: Int, precision: Int, tz_min: Int?, expectedErrorCode: ErrorCode) =
+        TimeTestCase(hour, min, sec, nano, precision, tz_min, expectedErrorCode = expectedErrorCode)
+
+    @Test
+    @Parameters
+    fun runTests(tc: TimeTestCase) {
+        when (tc.expectedErrorCode) {
+            null -> {
+                val time = Time.of(tc.hour, tc.min, tc.second, tc.nano, tc.precision, tc.tz_min)
+                assertEquals(tc.expectedLocalTime, time.localTime)
+                assertEquals(tc.expectedZoneOffset, time.zoneOffset)
+            }
+            else -> {
+                try {
+                    Time.of(tc.hour, tc.min, tc.second, tc.nano, tc.precision, tc.tz_min)
+                } catch (e: EvaluationException) {
+                    assertEquals(tc.expectedErrorCode, e.errorCode)
+                }
+            }
+        }
+    }
+
+    fun parametersForRunTests() = listOf(
+        case(
+            hour = 23,
+            min = 23,
+            sec = 12,
+            nano = 123456500,
+            precision = 6,
+            tz_min = null,
+            expectedLocalTime = LocalTime.of(23, 23, 12, 123456000),
+            expectedZoneOffset = null
+        ),
+        case(
+            hour = 23,
+            min = 23,
+            sec = 12,
+            nano = 123456600,
+            precision = 6,
+            tz_min = 300,
+            expectedLocalTime = LocalTime.of(23, 23, 12, 123457000),
+            expectedZoneOffset = ZoneOffset.ofTotalSeconds(300 * SECONDS_PER_MINUTE)
+        ),
+        case(
+            hour = 23,
+            min = 23,
+            sec = 12,
+            nano = 123456500,
+            precision = 9,
+            tz_min = 300,
+            expectedLocalTime = LocalTime.of(23, 23, 12, 123456500),
+            expectedZoneOffset = ZoneOffset.ofTotalSeconds(300 * SECONDS_PER_MINUTE)
+        ),
+        case(
+            hour = 23,
+            min = 23,
+            sec = 12,
+            nano = 123456789,
+            precision = 0,
+            tz_min = -18 * MINUTES_PER_HOUR,
+            expectedLocalTime = LocalTime.of(23, 23, 12, 0),
+            expectedZoneOffset = ZoneOffset.ofTotalSeconds( -18 * SECONDS_PER_HOUR)
+        ),
+        case(
+            hour = 23,
+            min = 23,
+            sec = 12,
+            nano = 123456789,
+            precision = 0,
+            tz_min = 18 * MINUTES_PER_HOUR,
+            expectedLocalTime = LocalTime.of(23, 23, 12, 0),
+            expectedZoneOffset = ZoneOffset.ofTotalSeconds( 18 * SECONDS_PER_HOUR)
+        ),
+        case(
+            hour = 23,
+            min = 23,
+            sec = 12,
+            nano = 123456789,
+            precision = 0,
+            tz_min = 18 * MINUTES_PER_HOUR,
+            expectedErrorCode = ErrorCode.EVALUATOR_DATETIME_EXCEPTION
+        ),
```

Comment:
It would be nice to have either a description field or comment indicating why this should fail (and other cases like it).  Not entirely obvious without looking at the case and even though I might infer an error, it might not be the same one you were thinking...

Reply:
Thanks a lot for pointing this out. This is indeed a happy case and should have failed saying "expected test to fail". I have modified the code.